src/app/projects/components/project-settings/helper/data-handler-helper.ts

```diff
@@ -84,20 +87,34 @@ export class DataHandlerHelper {
         const values = settingModals.embedding.create.embeddingCreationFormGroup.getRawValue();
         let toReturn = this.getAttributeArrayAttribute(values.targetAttribute, 'name', attributes);
         toReturn += "-" + (values.granularity == 'ON_ATTRIBUTE' ? 'classification' : 'extraction');
-        toReturn += "-" + values.embeddingHandle;
+        const platform = settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value;
+        if (platform == PlatformType.HUGGING_FACE || platform == PlatformType.PYTHON) {
+            toReturn += "-" + platform + "-" + values.model;
+        } else if (platform == PlatformType.OPEN_AI || platform == PlatformType.COHERE) {
+            toReturn += this.buildEmbeddingNameWithApiToken(values, platform);
+        }
+        return toReturn;
+    }
 
+    buildEmbeddingNameWithApiToken(values: any, platform: string) {
+        if (values.apiToken == null) return "";
+        let toReturn = "";
+        const apiTokenCut = values.apiToken.substring(0, 3) + "..." + values.apiToken.substring(values.apiToken.length - 4, values.apiToken.length);
+        toReturn += `-${platform}-${platform == PlatformType.OPEN_AI ? values.model + "-" + apiTokenCut : apiTokenCut}`;
```

Comment:
+= not necessary if nothing exists before

Also I don't like iterpolation strings with conditions inside since they are hard to read.

Maybe something like:

```suggestion
        if (values.apiToken == null) return "";
        const platformStr = "-" + platform + "-";
        const apiTokenCut = values.apiToken.substring(0, 3) + "..." + values.apiToken.substring(values.apiToken.length - 4, values.apiToken.length);
        if (platform == PlatformType.OPEN_AI) return platformStr + values.model + "-" + apiTokenCut;
        else return platformStr + apiTokenCut;
```